src/deploy/functions/prepare.ts

```diff
@@ -271,3 +272,29 @@ function maybeCopyTriggerRegion(wantE: backend.Endpoint, haveE: backend.Endpoint
   }
   wantE.eventTrigger.region = haveE.eventTrigger.region;
 }
+
+/** Figures out the blocking endpoint options by taking the OR of every trigger option and reassigning that value back to the endpoint. */
```

Comment:
Why is this the correct behavior? If I have a beforeUserCreated function I must have the same security options as a beforeUserLoggedIn function?

Reply:
It's part of the trigger proposal - http://go/cf3-auth-blocking-triggers

Since these are resource level options, they must be the same. Instead of erroring if a user sets the `beforeCreate` & `beforeSignIn` options to different values, we just take the OR of them. This will need to be made very clear in our docs